packages/core/src/node/logger-cli-contribution.ts

```
@@ -0,0 +1,142 @@
+    }
+
+    return level;
+}
+
+/** Maps logger names to log levels.  */
+export interface LogLevels {
+    [key: string]: LogLevel,
+}
+
+/**
+ * Parses command line switches related to log levels, then watches the log
+ * levels file (if specified) for changes.  This is the source of truth for
+ * what the log level per logger should be.
+ */
+@injectable()
+export class LogLevelCliContribution implements CliContribution {
+    logLevels: LogLevels = {};
+
+    /**
+     * Log level to use for loggers not specified in `logLevels`.
+     */
+    defaultLogLevel: LogLevel = LogLevel.INFO;
+
+    logLevelsChangeEvent: Emitter<{}> = new Emitter<{}>();
+
+    configure(conf: yargs.Argv): void {
+        conf.option('log-level', {
+            description: 'Sets the default log level',
+            choices: Array.from(LogLevel.strings.values()),
+            nargs: 1,
+        });
+
+        conf.option('log-levels-file', {
+            description: 'Path to a JSON file specyfing the log level of various loggers',
+            type: 'string',
+            nargs: 1,
+        });
+    }
+
+    async setArguments(args: yargs.Arguments): Promise<void> {
+        if (args['log-level'] !== undefined && args['log-levels-file'] !== undefined) {
+            throw new Error('--log-level and --log-levels-file are mutually exclusive.');
+        }
+
+        if (args['log-level'] !== undefined) {
+            this.defaultLogLevel = readLogLevelString(args['log-level'], 'Unknown log level passed to --log-level');
+        }
+
+        if (args['log-levels-file'] !== undefined) {
+            let filename = args['log-levels-file'];
+            filename = path.resolve(filename);
+
+            await this.slurpLogLevelsFile(filename);
+            await this.watchLogLevelsFile(filename);
+        }
+    }
+
+    watchLogLevelsFile(filename: string): Promise<void> {
+        return nsfw(filename, async (events: nsfw.ChangeEvent[]) => {
```

What about the case when a file does not exist? Will it interrupt the cli initialization?

---

The CLI initialization will have already been interrupted because `slurpLogLevelsFile` will have throw.

---

Note that this is the intended behavior.  If I open theia with `--log-levels-file=doesntexist.json`, I want it to fail starting, rather than silently fail.